api/lang/format.go

```diff
@@ -0,0 +1,30 @@
+package lang
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/getgauge/common"
+	"github.com/getgauge/gauge/formatter"
+	"github.com/getgauge/gauge/logger"
+	"github.com/getgauge/gauge/util"
+	"github.com/sourcegraph/go-langserver/pkg/lsp"
+	"github.com/sourcegraph/jsonrpc2"
+)
+
+func format(request *jsonrpc2.Request) (interface{}, error) {
+	var params lsp.TextDocumentPositionParams
+	if err := json.Unmarshal(*request.Params, &params); err != nil {
+		return nil, err
+	}
+	logger.APILog.Debugf("LangServer: request received : Type: Format Document URI: %s", params.TextDocument.URI)
+	file := convertURItoFilePath(params.TextDocument.URI)
+	if common.FileExists(file) && util.IsValidSpecExtension(file) {
+		res := formatter.FormatSpecFiles(file)
+		if len(res[0].ParseErrors) > 0 {
+			logger.APILog.Debugf("LangServer: Failed to format file : %s", params.TextDocument.URI)
+			return nil, errors.New(res[0].ParseErrors[0].Message)
+		}
+	}
+	return nil, nil
```

Comment:
I think Gauge need not do the formatting, it just has to return the textEdits describing the modifications.

That is what is described in the [protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#document-formatting-request) for Document Formatting Request's response.

> result: TextEdit[] describing the modification to the document to be formatted.